verkle-trie/src/database/meta.rs

```diff
@@ -11,86 +12,102 @@ pub struct StemMeta {
     pub stem_commitment: Element,
     pub hash_stem_commitment: Fr,
 }
-
 impl std::fmt::Debug for StemMeta {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("StemMeta")
-            .field("c_1", &hex::encode(compress_point_to_array(&self.c_1)))
-            .field("c_2", &hex::encode(compress_point_to_array(&self.c_2)))
-            .field("hash_c1", &hex::encode(scalar_to_array(&self.hash_c1)))
-            .field("hash_c2", &hex::encode(scalar_to_array(&self.hash_c2)))
+            .field(
+                "c_1",
+                &hex::encode(compress_point_to_array(&self.c_1).unwrap()),
+            )
+            .field(
+                "c_2",
+                &hex::encode(compress_point_to_array(&self.c_2).unwrap()),
+            )
+            .field(
+                "hash_c1",
+                &hex::encode(scalar_to_array(&self.hash_c1).unwrap()),
+            )
+            .field(
+                "hash_c2",
+                &hex::encode(scalar_to_array(&self.hash_c2).unwrap()),
+            )
             .field(
                 "stem commitment",
-                &hex::encode(compress_point_to_array(&self.stem_commitment)),
+                &hex::encode(compress_point_to_array(&self.stem_commitment).unwrap()),
             )
             .field(
                 "hash_stem_commitment",
-                &hex::encode(scalar_to_array(&self.hash_stem_commitment)),
+                &hex::encode(scalar_to_array(&self.hash_stem_commitment).unwrap()),
             )
             .finish()
     }
 }
 
-fn point_to_array(p: &Element) -> [u8; 64] {
+use ark_serialize::CanonicalSerialize;
+
+fn point_to_array(p: &Element) -> Result<[u8; 64], SerializationError> {
     let mut bytes = [0u8; 64];
-    use ark_serialize::CanonicalSerialize;
-    p.serialize_uncompressed(&mut bytes[..]).unwrap();
-    bytes
+    p.serialize_uncompressed(&mut bytes[..])?;
```

Comment:
I'm somewhat a bit uncertain about this because if the library is no longer able to serialise a point, then theres no recovery from that and the node should stop immediately. There is no clear decision on whether this should indeed propagate an error, so I'll leave it as is since it makes it usable in other non-node software whom may never want to panic

Reply:
My main thought here had been similar, that if serialization fails its unrecoverable, but I think we might also want to consider how the initial "Verge" migration might have a scenario where someone's node is migrating their entire DB in the background, hits an out of space issue, and the node might either want to more aggressively delete old Merkle content, or atleast safely halt so that progress isn't lost